lib/srv/discovery/discovery.go

```diff
@@ -1217,17 +1237,19 @@ func (s *Server) startDynamicWatcherUpdater() {
 					// If the user updates the DiscoveryGroup to DG2, then DC1 must be removed from the scope of this process.
 					// We blindly delete it, in the worst case, this is a no-op.
 					s.deleteDynamicFetchers(event.Resource.GetName())
+					s.triggerPoll()
 					continue
 				}
 
 				if err := s.upsertDynamicMatchers(dc); err != nil {
 					s.Log.WithError(err).Warnf("failed to update dynamic matchers for discovery config %q", dc.GetName())
 					continue
 				}
+				s.triggerPoll()
 
 			case types.OpDelete:
 				s.deleteDynamicFetchers(event.Resource.GetName())
-
+				s.triggerPoll()
```

Comment:
In this scope "poll" context is unclear for reader. 
Maybe just notifyDiscoveryConfigChanged